src/lib/syncDocsPath.ts

```diff
@@ -127,6 +130,40 @@ async function pushDoc(
     });
 }
 
+function sortFiles(files: string[], { allowedFileExtensions }: { allowedFileExtensions: string[] }): string[] {
+  const extensionRegexp = new RegExp(`(${allowedFileExtensions.join('|')})$`);
+
+  const filesBySlug = files
+    .map(filePath => {
+      const doc = frontMatter(fsSync.readFileSync(filePath, 'utf8'));
+      const slug = path.basename(filePath).replace(extensionRegexp, '');
+      const parentDocSlug = doc.data.parentDocSlug;
+
+      return { filePath, slug, parentDocSlug };
+    })
+    .reduce(
+      (bySlug, obj) => {
+        // eslint-disable-next-line no-param-reassign
+        bySlug[obj.slug] = obj;
+        return bySlug;
+      },
+      {} as Record<string, { filePath: string; parentDocSlug: string; slug: string }>,
+    );
```

Comment:
(forgive me if this is slightly off from a syntax/formatting standpoint) would something like this work?

```suggestion
    .reduce<Record<string, { filePath: string; parentDocSlug: string; slug: string }>>(
      (bySlug, obj) => {
        // eslint-disable-next-line no-param-reassign
        bySlug[obj.slug] = obj;
        return bySlug;
      },
      {},
    );
```